torch_xla/csrc/tensor.cpp

```diff
@@ -1281,7 +1308,7 @@ void XLATensor::SyncLiveTensorsGraph(const Device* device,
 
 void XLATensor::MarkStep(const Device* device) {
   XLA_COUNTER("MarkStep", 1);
-  DeviceContextArena::Get()->ClearProfileData(device);
+  DeviceContextArena::Get()->StepRngSeed(device);
```

Comment:
wondering why is this needed. Even without this, everytime someone call `GetRngSeed`, they will get a new seed right?

Reply:
This has the same reason of `mark_step()` for tensors.
If we do not "trim" the IR chain, the seed IR graph keeps growing from the master seed (device data) down to its current head.